### PR TITLE
Adds project-name as an optional argument for project initialization

### PR DESCRIPTION
**Pull request by schnie**
## Description

This PR just allows the project name to be specified as an optional, positional argument. When used this way, we set the argument as the project name, and then additionally create a directory with the same name, and initialize a fresh project in that new directory. This change is purely additive, and existing commands and flags work the same as before.

### Current
```
mkdir test-project
cd test-project
astro dev init --name test-project
```
### New
```
astro dev init test-project
cd test-project
```

## Example Usage

This change specifically:
```
❯ astro dev init etl
Initialized empty Astro project in /Users/schnie/repos/etl
To begin developing, change to your project directory with `cd etl`
```

Combined with https://github.com/astronomer/astro-cli/pull/1728:
```
❯ astro dev init etl --from-template=etl
Initialized empty Astro project in /Users/schnie/repos/etl
To begin developing, change to your project directory with `cd etl`
```

## 🎟 Issue(s)

Resolves to https://github.com/astronomer/astro/issues/25473

## 🧪 Functional Testing

```
❯ astro dev init test-project
Initialized empty Astro project in /Users/schnie/repos/test-project
To begin developing, change to your project directory with `cd /Users/schnie/repos/test-project`
```

## 📸 Screenshots

> Add screenshots to illustrate the validity of these changes.

## 📋 Checklist

- [x] Rebased from the main (or release if patching) branch (before testing)
- [x] Ran `make test` before taking out of draft
- [x] Ran `make lint` before taking out of draft
- [x] Added/updated applicable tests
- [x] Tested against [Astro-API](https://github.com/astronomer/astro/) (if necessary).
- [x] Tested against [Houston-API](https://github.com/astronomer/houston-api/) and [Astronomer](https://github.com/astronomer/astronomer/) (if necessary).
- [ ] Communicated to/tagged owners of respective clients potentially impacted by these changes.
- [ ] Updated any related [documentation](https://github.com/astronomer/docs/)

### CODE REVIEW

File: cmd/airflow.go [hunk 1]
```diff
@@ -491,12 +492,18 @@ func newObjectExportCmd() *cobra.Command {
...
+		projectName = args[0]
+		config.WorkingPath = filepath.Join(config.WorkingPath, projectName)
+	}
+
 	// Validate project name
 	if projectName != "" {
-		// error if project name has spaces
-		if len(args) > 0 {
-			return errProjectNameSpaces
-		}
```
Comment (schnie) [inline cmd/airflow.go:499]: This isn't needed because we already use cobra's NumberOfArguments configuration on this command. Anything over 1 argument shows the help message so this condition is never reached.

File: cmd/airflow.go [hunk 2]
```diff
@@ -529,24 +535,39 @@ func airflowInit(cmd *cobra.Command, args []string) error {
...
 			fmt.Println("Canceling project initialization...")
 			return nil
 		}
 	}
 
-	exists := config.ProjectConfigExists()
+	exists, err := config.IsProjectDir(config.WorkingPath)
+	if err != nil {
+		return err
+	}
```
Comment (schnie) [inline cmd/airflow.go:570]: Alternative function we already had to check if the project config file exists in the directory or not. The prior version assumes that the process is already within the working directory, where the alternative lets us check if the file exists at the certain path.

File: cmd/airflow.go [hunk 3]
```diff
@@ -554,9 +594,16 @@ func airflowInit(cmd *cobra.Command, args []string) error {
 	}
 
 	if exists {
-		fmt.Printf(configReinitProjectConfigMsg+"\n", config.WorkingPath)
+		fmt.Printf(configReinitProjectConfigMsg, config.WorkingPath)
 	} else {
-		fmt.Printf(configInitProjectConfigMsg+"\n", config.WorkingPath)
+		fmt.Printf(configInitProjectConfigMsg, config.WorkingPath)
+	}
```
Comment (schnie) [inline cmd/airflow.go:600]: Just standardized the messages and the format strings and fixed a double vs single newline inconsistency.

File: cmd/airflow.go [hunk 4]
```diff
@@ -517,24 +527,54 @@ func airflowInit(cmd *cobra.Command, args []string) error {
 	defaultImageName := airflow.AstroRuntimeImageName
 	if useAstronomerCertified {
 		defaultImageName = airflow.AstronomerCertifiedImageName
-		fmt.Printf("Initializing Astro project\nPulling Airflow development files from Astronomer Certified Airflow Version %s\n", defaultImageTag)
-	} else {
-		fmt.Printf("Initializing Astro project\nPulling Airflow development files from Astro Runtime %s\n", defaultImageTag)
```
Comment (schnie) [inline cmd/airflow.go:522]: These messages are not necessary and duplicative. I've also updated the corresponding test for this output.

File: cmd/errors.go [hunk 5]
```diff
@@ -10,7 +10,6 @@ var (
 	errInvalidBothCustomImageandVersion            = errors.New("you provided both a Custom image and a version. You have to provide only one of these to upgrade")                             //nolint
 
 	errConfigProjectName = errors.New("project name is invalid")
-	errProjectNameSpaces = errors.New("this project name is invalid, a project name cannot contain spaces. Try using '-' instead")
```
Comment (schnie) [inline cmd/errors.go:13]: Obsolete error message. Cobra will show help text if more than one argument is specified.

File: cmd/airflow.go [hunk 6]
```diff
@@ -463,12 +466,21 @@ func newObjectExportCmd() *cobra.Command {
 
 // Use project name for image name
 func airflowInit(cmd *cobra.Command, args []string) error {
+	// The first positional argument is the project name.
+	// If the project name is provided in this way, we'll
+	// attempt to create a directory with that name.
+	if len(args) > 0 {
+		projectName = args[0]
+	}
```
Comment (neel-astro) [inline cmd/airflow.go:474]: This might mean that if someone is trying `astro dev init --name test project`, would result in a project name as `project` when they might be looking for `test project`, should we at least warn and confirm if not error, when someone is setting both the flag and arg?

**Review by neel-astro — COMMENTED** [review]
Looks fine apart from one minor edge case :)

File: cmd/airflow.go [hunk 7]
```diff
@@ -554,14 +559,80 @@ func airflowInit(cmd *cobra.Command, args []string) error {
...
+	// Return the path we just created.
+	return newProjectPath, nil
+}
+
+func ensureProjectName(args []string, projectName string) (string, error) {
+	// If the project name is specified with the --name flag,
+	// it cannot be specified as a positional argument as well, so return an error.
+	if projectName != "" && len(args) > 0 {
+		return "", errConfigProjectNameSpecifiedTwice
+	}
```
Comment (schnie) [inline cmd/airflow.go:610]: @neel-astro added this case here to throw an error message if both the flag and positional argument are used.

I also separated all the new and existing "project name" logic into this new function to make the linter happy. :)

File: cmd/airflow.go (hunk 6 above)
Comment (schnie) [reply cmd/airflow.go]: Good catch. Updated the logic here to catch this case and throw an error.

Comment (schnie) [reply cmd/airflow.go]: Also added a corresponding test.

**Review by neel-astro — APPROVED** [review]

**Review by pritt20 — APPROVED** [review]
